### PR TITLE
Django upgrade

### PR DESCRIPTION
**Pull request by blip2**
Updates django to latest version
Provides dockerfile for simple development environment
Adds search to admin permissions form
cleaned up various legacy/incomplete code

### CODE REVIEW

**Comment by blip2** [comment]
added pip venv

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+RUN apt-get update && apt-get install -y libsasl2-dev libldap2-dev libssl-dev && rm -rf /var/lib/apt/lists/*
+RUN python3 -m venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
+RUN pip install --upgrade pip
+COPY ./requirements.txt ./
+RUN pip install -r requirements.txt
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+COPY ./ /server/
+RUN python manage.py collectstatic --noinput
```
Comment (mich181189) [inline Dockerfile:13]: Don't you want an entrypoint/cmd line?

File: README.md [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
-## For LDAP syncing:
+To create an initial user for testing:
 
 ```
-pip install --verbose --no-deps django-ldap-sync
+docker-compose exec server python manage.py createsuperuser
 ```
 
-To setup the db run:
+## Legacy
```
Comment (mich181189) [inline README.md]: I really don't like calling things legacy when they're just an alternative method the author didn't like :-P

I think in many cases, just using virtualenv will be enough - I don't really see what Docker adds to this. Nothing wrong with it being an option, but I don't think I'd call it "legacy"

**Review by mich181189 — APPROVED** [review]

**Comment by blip2** [comment]
@mich181189 updated but I have no access to merge

**Comment by mich181189** [comment]
> @mich181189 updated but I have no access to merge

Should do now

**Pull request merged**